src/settings/toml/builder.rs

```diff
@@ -0,0 +1,66 @@
+use std::env;
+use std::path::PathBuf;
+use std::process::Command;
+
+use serde::{Deserialize, Serialize};
+
+use crate::terminal::message::{Message, StdOut};
+
+use super::ScriptFormat;
+
+const BUILD_DIR: &str = "dist";
+const SRC_DIR: &str = "src";
+
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct Builder {
+    pub build_command: Option<String>,
+    #[serde(default = "build_dir")]
+    pub build_dir: PathBuf,
+    pub upload_format: ScriptFormat,
+    #[serde(default = "src_dir")]
+    pub src_dir: PathBuf,
+}
+
+fn default_warning(field: &str, default: &str) {
+    StdOut::warn(&format!(
+        "{} not specified, falling back to {}",
+        field, default
+    ));
+}
+
+fn build_dir() -> PathBuf {
+    default_warning("build dir", BUILD_DIR);
```

Comment:
It's a little confusing to warn on build dir and src dir when they aren't actually used if upload_format is set to service-worker, right?

Reply:
I'll address this when I address @ags799 concern below about having src/build dir be `./`, we don't want to allow that or allow empty directories.

Reply:
I ended up removing the default warning. I'll add a separate validation for the `./` and nonexistent directory cases